feat(aft37): add payable mint extension

ThomasMarches opened the pull request.

ThomasMarches committed:
feat(aft37): add payable mint extension

SailorSnoW reviewed (changes requested):
seems good, some warnings need to be fixed tho

ThomasMarches committed:
fix(aft37-payable-mint): remove warnings

feat(aft37): change max_supply type

feat(aft37): change max_supply type

fix(aft37-payable-mint): max_supply type

fix(aft37-payable-mint): CollectionIsFull condition

ThomasMarches committed:
wip: add total_balance to payable_mint extension
